pkg/sbom/node.go

```diff
@@ -240,7 +240,10 @@ func (n *Node) flatString() string {
 			for i := 0; i < v.List().Len(); i++ {
 				pairs = append(pairs, fmt.Sprintf("%s[%d]:%s", fd.FullName(), i, v.List().Get(i)))
 			}
-
+		case "bomsquad.protobom.Node.primary_purpose":
+			for _, p := range n.PrimaryPurpose {
+				pairs = append(pairs, fmt.Sprintf("primary_purpose:%s", p.String()))
+			}
```

Comment:
I added a specific check for bomsquad.protobom.Node.primary_purpose into flatStringStrSlice() so we get the name of the enum value instead of the number.  I think that will be nicer to work with, and our tests will survive if the enum values change in the future.  Hope you don't mind what I did there.

Reply:
After discussing this a little bit we decided not to include the special case.